public/docs/dart/latest/guide/dependency-injection.jade

```diff
@@ -1,279 +1,138 @@
-+includeShared('{ts}', 'providers-factory-3')
-+makeExample('dependency-injection/dart/lib/heroes/hero_service_provider.dart','provider', 'lib/heroes/hero_service_provider.dart (excerpt)')(format='.')
-+includeShared('{ts}', 'providers-factory-4')
-+includeShared('{ts}', 'providers-factory-5')
-- var stylePattern = { otl: /(providers.*),$/gm };
-+makeTabs(
-  `dependency-injection/dart/lib/heroes/heroes_component.dart,
-  dependency-injection/dart/lib/heroes/heroes_component_1.dart`,
-  null,
-  `lib/heroes/heroes_component (v.3),
-  lib/heroes/heroes_component (v.2)`,
-  stylePattern)
-+includeShared('{ts}', 'tokens-1')
-+makeExample('dependency-injection/dart/lib/injector_component.dart','get-hero-service')(format='.')
-+includeShared('{ts}', 'tokens-2')
-// [PENDING: How about a better name than ProviderComponent8?]
-+makeExample('dependency-injection/dart/lib/providers_component.dart','provider-8-ctor')(format=".")
-+includeShared('{ts}', 'tokens-3')
-.callout.is-helpful
-  header Dart difference: Interfaces are valid tokens
   :marked
-    In TypeScript, interfaces don't work as provider tokens.
-    Dart doesn't have this problem;
-    every class implicitly defines an interface,
-    so interface names are just class names.
-+includeShared('{ts}', 'tokens-non-class-deps-1')
-:marked
-  We know we can register an object with a [value provider](#value-provider).
-  But what do we use for the token?
-+includeShared('{ts}', 'tokens-opaque-1')
-+makeExample('dependency-injection/dart/lib/providers_component.dart','opaque-token')(format='.')
-+includeShared('{ts}', 'tokens-opaque-2')
-+makeExample('dependency-injection/dart/lib/providers_component.dart','use-opaque-token')(format=".")
-:marked
-  Here's an example of providing configuration information
-  for an injected class. First define the class:
-+makeExample('dependency-injection/dart/lib/providers_component.dart','configurable-logger')(format=".")
-:marked
-  Then inject that class and its configuration information:
-+makeExample('dependency-injection/dart/lib/providers_component.dart','providers-7')(format=".")
+    The problem with using `Map` as a token, is not that it is an abstract
+    class, but rather that it is very general purpose (like `String`) and so
+    our application may depend on several maps each used for a
+    different purpose.
 
+block dart-map-alternative
+  :marked
+    As an alternative to using a configuration `Map`, we can define
+    a custom class type:
 
-+includeShared('{ts}', 'summary')
+  +makeExample('dependency-injection/ts/app/app.config.ts','config-alt','app/app-config.ts (alternative config)')(format='.')
 
-+includeShared('{ts}', 'appendix-explicit-injector-1')
-+makeExample('dependency-injection/dart/lib/injector_component.dart', 'injector', 'lib/injector_component.dart')
-+includeShared('{ts}', 'appendix-explicit-injector-2')
+  :marked
+    One of the key benefits of defining a configuration object class, is
```

Comment:
Move the comma to be before "relative". Maybe change "relative..." to be clearer. I'm not sure if you're comparing the map & class benefits, or something else.

(Where the comma is, it's equivalent to "This, is that", which is you wouldn't normally do. If the sentence seems too hard to read without the comma, I'd rewrite to make it clearer. Maybe: Defining a configuration object class has a few benefits. One key benefit is strong static checking: we'll be warned early if we misspell a property name or assign it a value of the wrong type.)